cpp/arcticdb/storage/python_bindings.cpp

```diff
@@ -70,16 +70,32 @@ void register_bindings(py::module& storage, py::exception<arcticdb::ArcticExcept
         .value("WRITE", OpenMode::WRITE)
         .value("DELETE", OpenMode::DELETE);
 
-    storage.def("create_library_index", &create_library_index);
+    py::class_<AzureBaseCredential, std::shared_ptr<AzureBaseCredential>>(storage, "AzureBaseCredential");
+    py::class_<Azure::Identity::DefaultAzureCredential, std::shared_ptr<Azure::Identity::DefaultAzureCredential>, AzureBaseCredential>(storage, "AzureDefaultCredential")
+        .def(py::init<>());
 
-    storage.def("create_mem_config_resolver", [](const py::object & env_config_map_py) -> std::shared_ptr<ConfigResolver> {
+    py::class_<StorageCredential>(storage, "StorageCredential")
+        .def(py::init<>())
+        .def(py::pickle(
```

Comment:
Confused - why do we need / want this object to be pickleable? PIckling should make us afraid, very afraid